### ISSUE TITLE
Program does not exit after parsing --help or --version command line options

### ISSUE THREAD

**Issue by jcjc29** [RPCS3/rpcs3#5642]
Using the v0.0.6-7790 (19ff95da) Linux AppImage, RPCS3 hangs after printing the help or version information instead of exiting.

```
$ ./rpcs3-v0.0.6-7790-19ff95da_linux64.AppImage --help
Usage: rpcs3 [options] (S)ELF [Args...]
Welcome to RPCS3 command line.

Options:
  -h, --help     Displays this help.
  -v, --version  Displays version information.

Arguments:
  (S)ELF         Path for directly executing a (S)ELF
  [Args...]      Optional args for the executable

...
*hangs forever*
```

**Comment by AniLeo** [comment]
I've investigated this:

If main returns before `app.Init()`, `s_gui_listener` deallocation throws an exception (xmemory0 line 207) and process turns into a zombie process.

Needs to be properly destructed.

Tested on Windows too, it's not an OS specific issue.

**Comment by AniLeo** [comment]
Fixed by #5718
